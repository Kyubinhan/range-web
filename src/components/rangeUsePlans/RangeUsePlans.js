import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import debounce from 'lodash.debounce';
import { Button, Modal, Header } from 'semantic-ui-react';

import RangeUsePlansTable from './RangeUsePlansTable';
import RangeUsePlansSearch from './RangeUsePlansSearch';

import { getMockRangeUsePlans } from './test/mockValues';

const propTypes = {
  rangeUsePlans: PropTypes.array.isRequired,
  searchRangeUsePlans: PropTypes.func.isRequired,
};

const defaultProps = {
  rangeUsePlans: getMockRangeUsePlans(7),
  searchRangeUsePlans: (term) => {
    console.log(term);
  },
};

export class RangeUsePlans extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
    // wait for 1 second for the user to finish writing a search term
    // then make a network call
    this.searchRangeUsePlans = debounce(props.searchRangeUsePlans, 1000);
  }  
  
  handleSearchInput = (searchTerm) => {
    this.searchRangeUsePlans(searchTerm)
  }
  
  render() {
    const { rangeUsePlans } = this.props;

    return (
      <div className="range-use-plans">
        <Header as='h1'>Range Use Plans</Header>

        <div className="range-use-plans__actions">
          <div className="range-use-plans__assign">
            <Modal trigger={<Button primary>Assign Staffs</Button>}>
              <Modal.Header>Select Assignee</Modal.Header>
              <Modal.Content>
                <Modal.Description>
                  <Header>Assign an officer to a tenure</Header>
                  <p>Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.</p>
                </Modal.Description>
              </Modal.Content>
            </Modal>
          </div>

          <div className="range-use-plans__search">
            <RangeUsePlansSearch
              label="Search Agreements:"
              placeholder="Enter Search Terms..." 
              handleSearchInput={this.handleSearchInput}
            />
          </div>
        </div>

        <RangeUsePlansTable 
          rangeUsePlans={rangeUsePlans}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {

  }; 
};

RangeUsePlans.propTypes = propTypes;
RangeUsePlans.defaultProps = defaultProps;

export default connect(
  mapStateToProps, null
)(RangeUsePlans);