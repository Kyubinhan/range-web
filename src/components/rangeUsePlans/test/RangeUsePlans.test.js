import React from 'react';
import { shallow } from 'enzyme';
import { RangeUsePlans } from '../RangeUsePlans';

const props = {};
const setupProps = () => {
  props.rangeUsePlans = [];
  props.searchRangeUsePlans = jest.fn();
};

beforeEach(() => {
  setupProps();
});

describe('RangeUsePlans', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RangeUsePlans {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `RangeUsePlanSearch and RangeUsePlanTable`', () => {
    const wrapper = shallow(<RangeUsePlans {...props} />);
    expect(wrapper.find('RangeUsePlanSearch').exists()).toBe(true);    
    expect(wrapper.find('RangeUsePlanTable').exists()).toBe(true);    
  });

  describe('Event handlers', () => {
    it('`handleSearchInput` calls searchRangeUsePlans function', () => {
      const wrapper = shallow(<RangeUsePlans {...props} />);
      const instance = wrapper.instance();
      const mockSearchTerm = 'search';
      const searchTenureAgreementsSpy = jest.spyOn(instance, 'searchRangeUsePlans');
      
      instance.handleSearchInput(mockSearchTerm);
      expect(searchTenureAgreementsSpy).toHaveBeenCalledWith(mockSearchTerm);
    });
  });
});