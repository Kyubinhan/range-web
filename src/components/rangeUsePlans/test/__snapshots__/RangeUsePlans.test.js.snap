// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeUsePlans renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RangeUsePlans
    rangeUsePlans={Array []}
    searchRangeUsePlans={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
>
          Range Use Plans
</Header>,
        <div
          className="tenure-agreement__actions"
>
          <div
                    className="tenure-agreement__assign"
          >
                    <Modal
                              closeOnDimmerClick={true}
                              closeOnDocumentClick={false}
                              dimmer={true}
                              eventPool="Modal"
                              trigger={
                                        <Button
                                                  as="button"
                                                  primary={true}
                                        >
                                                  Assign Staffs
                                        </Button>
                              }
                    >
                              <ModalHeader>
                                        Select Assignee
                              </ModalHeader>
                              <ModalContent>
                                        <ModalDescription>
                                                  <Header>
                                                            Assign an officer to a tenure
                                                  </Header>
                                                  <p>
                                                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                  </p>
                                        </ModalDescription>
                              </ModalContent>
                    </Modal>
          </div>
          <div
                    className="tenure-agreement__search"
          >
                    <RangeUsePlanSearch
                              handleSearchInput={[Function]}
                              label="Search Agreements:"
                              placeholder="Enter Search Terms..."
                    />
          </div>
</div>,
        <RangeUsePlanTable
          rangeUsePlans={Array []}
/>,
      ],
      "className": "tenure-agreement",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": "Range Use Plans",
        },
        "ref": null,
        "rendered": "Range Use Plans",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="tenure-agreement__assign"
>
              <Modal
                            closeOnDimmerClick={true}
                            closeOnDocumentClick={false}
                            dimmer={true}
                            eventPool="Modal"
                            trigger={
                                          <Button
                                                        as="button"
                                                        primary={true}
                                          >
                                                        Assign Staffs
                                          </Button>
                            }
              >
                            <ModalHeader>
                                          Select Assignee
                            </ModalHeader>
                            <ModalContent>
                                          <ModalDescription>
                                                        <Header>
                                                                      Assign an officer to a tenure
                                                        </Header>
                                                        <p>
                                                                      Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                        </p>
                                          </ModalDescription>
                            </ModalContent>
              </Modal>
</div>,
            <div
              className="tenure-agreement__search"
>
              <RangeUsePlanSearch
                            handleSearchInput={[Function]}
                            label="Search Agreements:"
                            placeholder="Enter Search Terms..."
              />
</div>,
          ],
          "className": "tenure-agreement__actions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Modal
                closeOnDimmerClick={true}
                closeOnDocumentClick={false}
                dimmer={true}
                eventPool="Modal"
                trigger={
                                <Button
                                                as="button"
                                                primary={true}
                                >
                                                Assign Staffs
                                </Button>
                }
>
                <ModalHeader>
                                Select Assignee
                </ModalHeader>
                <ModalContent>
                                <ModalDescription>
                                                <Header>
                                                                Assign an officer to a tenure
                                                </Header>
                                                <p>
                                                                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                </p>
                                </ModalDescription>
                </ModalContent>
</Modal>,
              "className": "tenure-agreement__assign",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <ModalHeader>
                    Select Assignee
</ModalHeader>,
                  <ModalContent>
                    <ModalDescription>
                                        <Header>
                                                            Assign an officer to a tenure
                                        </Header>
                                        <p>
                                                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                        </p>
                    </ModalDescription>
</ModalContent>,
                ],
                "closeOnDimmerClick": true,
                "closeOnDocumentClick": false,
                "dimmer": true,
                "eventPool": "Modal",
                "trigger": <Button
                  as="button"
                  primary={true}
>
                  Assign Staffs
</Button>,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Select Assignee",
                  },
                  "ref": null,
                  "rendered": "Select Assignee",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <ModalDescription>
                      <Header>
                                            Assign an officer to a tenure
                      </Header>
                      <p>
                                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                      </p>
</ModalDescription>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        <Header>
                          Assign an officer to a tenure
</Header>,
                        <p>
                          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
</p>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": "Assign an officer to a tenure",
                        },
                        "ref": null,
                        "rendered": "Assign an officer to a tenure",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                        },
                        "ref": null,
                        "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                        "type": "p",
                      },
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RangeUsePlanSearch
                handleSearchInput={[Function]}
                label="Search Agreements:"
                placeholder="Enter Search Terms..."
/>,
              "className": "tenure-agreement__search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleSearchInput": [Function],
                "label": "Search Agreements:",
                "placeholder": "Enter Search Terms...",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "rangeUsePlans": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
>
            Range Use Plans
</Header>,
          <div
            className="tenure-agreement__actions"
>
            <div
                        className="tenure-agreement__assign"
            >
                        <Modal
                                    closeOnDimmerClick={true}
                                    closeOnDocumentClick={false}
                                    dimmer={true}
                                    eventPool="Modal"
                                    trigger={
                                                <Button
                                                            as="button"
                                                            primary={true}
                                                >
                                                            Assign Staffs
                                                </Button>
                                    }
                        >
                                    <ModalHeader>
                                                Select Assignee
                                    </ModalHeader>
                                    <ModalContent>
                                                <ModalDescription>
                                                            <Header>
                                                                        Assign an officer to a tenure
                                                            </Header>
                                                            <p>
                                                                        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                            </p>
                                                </ModalDescription>
                                    </ModalContent>
                        </Modal>
            </div>
            <div
                        className="tenure-agreement__search"
            >
                        <RangeUsePlanSearch
                                    handleSearchInput={[Function]}
                                    label="Search Agreements:"
                                    placeholder="Enter Search Terms..."
                        />
            </div>
</div>,
          <RangeUsePlanTable
            rangeUsePlans={Array []}
/>,
        ],
        "className": "tenure-agreement",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": "Range Use Plans",
          },
          "ref": null,
          "rendered": "Range Use Plans",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="tenure-agreement__assign"
>
                <Modal
                                closeOnDimmerClick={true}
                                closeOnDocumentClick={false}
                                dimmer={true}
                                eventPool="Modal"
                                trigger={
                                                <Button
                                                                as="button"
                                                                primary={true}
                                                >
                                                                Assign Staffs
                                                </Button>
                                }
                >
                                <ModalHeader>
                                                Select Assignee
                                </ModalHeader>
                                <ModalContent>
                                                <ModalDescription>
                                                                <Header>
                                                                                Assign an officer to a tenure
                                                                </Header>
                                                                <p>
                                                                                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                                </p>
                                                </ModalDescription>
                                </ModalContent>
                </Modal>
</div>,
              <div
                className="tenure-agreement__search"
>
                <RangeUsePlanSearch
                                handleSearchInput={[Function]}
                                label="Search Agreements:"
                                placeholder="Enter Search Terms..."
                />
</div>,
            ],
            "className": "tenure-agreement__actions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Modal
                  closeOnDimmerClick={true}
                  closeOnDocumentClick={false}
                  dimmer={true}
                  eventPool="Modal"
                  trigger={
                                    <Button
                                                      as="button"
                                                      primary={true}
                                    >
                                                      Assign Staffs
                                    </Button>
                  }
>
                  <ModalHeader>
                                    Select Assignee
                  </ModalHeader>
                  <ModalContent>
                                    <ModalDescription>
                                                      <Header>
                                                                        Assign an officer to a tenure
                                                      </Header>
                                                      <p>
                                                                        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                                      </p>
                                    </ModalDescription>
                  </ModalContent>
</Modal>,
                "className": "tenure-agreement__assign",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <ModalHeader>
                      Select Assignee
</ModalHeader>,
                    <ModalContent>
                      <ModalDescription>
                                            <Header>
                                                                  Assign an officer to a tenure
                                            </Header>
                                            <p>
                                                                  Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                                            </p>
                      </ModalDescription>
</ModalContent>,
                  ],
                  "closeOnDimmerClick": true,
                  "closeOnDocumentClick": false,
                  "dimmer": true,
                  "eventPool": "Modal",
                  "trigger": <Button
                    as="button"
                    primary={true}
>
                    Assign Staffs
</Button>,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Select Assignee",
                    },
                    "ref": null,
                    "rendered": "Select Assignee",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <ModalDescription>
                        <Header>
                                                Assign an officer to a tenure
                        </Header>
                        <p>
                                                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
                        </p>
</ModalDescription>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          <Header>
                            Assign an officer to a tenure
</Header>,
                          <p>
                            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
</p>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": "Assign an officer to a tenure",
                          },
                          "ref": null,
                          "rendered": "Assign an officer to a tenure",
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                          },
                          "ref": null,
                          "rendered": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
                          "type": "p",
                        },
                      ],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <RangeUsePlanSearch
                  handleSearchInput={[Function]}
                  label="Search Agreements:"
                  placeholder="Enter Search Terms..."
/>,
                "className": "tenure-agreement__search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "handleSearchInput": [Function],
                  "label": "Search Agreements:",
                  "placeholder": "Enter Search Terms...",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "rangeUsePlans": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
